pkg/tracing/otel_exporters.go

```diff
@@ -0,0 +1,67 @@
+package tracing
+
+import (
+	"context"
+	"time"
+
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/log"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+)
+
+type OtelExporterSettings struct {
+	Endpoint    string        `cfg:"endpoint" default:"localhost:4318"`
+	UrlPath     string        `cfg:"url_path" default:"/v1/traces"`
+	Compression bool          `cfg:"compression" default:"true"`
+	Insecure    bool          `cfg:"insecure" default:"false"`
+	Timeout     time.Duration `cfg:"timeout" default:"10s"`
+	Retry       RetryConfig   `cfg:"retry"`
+}
+
+type RetryConfig struct {
+	Enabled         bool          `cfg:"enabled" default:"false"`
+	InitialInterval time.Duration `cfg:"initial_interval"`
+	MaxInterval     time.Duration `cfg:"max_interval"`
+	MaxElapsedTime  time.Duration `cfg:"max_elapsed_time"`
+}
+
+type OtelExporterFactory func(ctx context.Context, config cfg.Config, logger log.Logger) (*otlptrace.Exporter, error)
+
+func AddTraceExporter(name string, exporter OtelExporterFactory) {
+	TraceExporters[name] = exporter
+}
+
+var TraceExporters = map[string]OtelExporterFactory{
```

Comment:
I think this shouldn't be exported. Also, I think this should be called `otelTraceExporters` as it seems OTEL specific.

Reply:
you are right, it is OTEL specific and yeah it doesn't have to be exported because if someone wants to add their implementation they can use the `AddTraceExporter` func (which I'll also rename to AddOtelTraceExporter)